airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/http_requester.py

```diff
@@ -100,20 +102,17 @@ def request_headers(
     def request_body_data(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
     ) -> Optional[Union[Mapping, str]]:
-        # FIXME: this should be declarative
-        return dict()
+        return self._request_options_provider.request_body_data(stream_state, stream_slice, next_page_token)
 
     def request_body_json(
         self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
     ) -> Optional[Mapping]:
-        # FIXME: this should be declarative
-        return dict()
+        return self._request_options_provider.request_body_json(stream_state, stream_slice, next_page_token)
```

Comment:
@brianjlai should we also add the POST method to `HttpMethod` while we're at it so we can knock off https://github.com/airbytehq/airbyte/issues/12884 at the same time?

Reply:
@brianjlai I missed that. looks good then!